### DISCUSSION TITLE
Spoofing multiple answers in dnsdist is randomized

### DISCUSSION THREAD

**Discussion by rizkymma** [PowerDNS/pdns#14737] [Q&A] [answered]
Hello,

I have dnsdist ver. 1.9.6-1pdns.jammy and tried to modify/spoof the answer, but the spoofed answer order is randomized:

```
# dig @127.0.0.1 google.com +short
2.2.2.2
1.1.1.1
3.3.3.3
# dig @127.0.0.1 google.com +short
3.3.3.3
1.1.1.1
2.2.2.2
# dig @127.0.0.1 google.com +short
1.1.1.1
3.3.3.3
2.2.2.2
```

```
function inputFunction(dq)
  local packet = dq:getContent()
  local overlay = newDNSPacketOverlay(packet)
  local cached = getPool('mycache'):getCache():getAddressListByDomain(overlay.qname)
  if table.getn(cached)>1 then
    -- dq:spoof(cached)
    return DNSAction.Spoof, '1.1.1.1,2.2.2.2,3.3.3.3'
  else
    return DNSAction.None
  end
end

addAction(AllRule(), LuaAction(inputFunction))
```

I also tried using dq:spoof() and it also returns answers in randomized order, is this expected?

Thanks in advance

**Comment by rgacogne** [comment, marked as answer]
Hi,

> I also tried using dq:spoof() and it also returns answers in randomized order, is this expected?

Yes, this is the expected behaviour, since this is generally what people want. If you tell us why you do not want that behaviour we might be able to help you.

**Reply by rizkymma** [reply]
Thanks for confiriming!

I'm trying to make it return a round-robin balanced responses, so that every answer has equal load instead of just random.

**Reply by rgacogne** [reply]
Right, then I believe the current behaviour is the one you want, since most clients are only going to look at the first record in the answer anyway. If you need more control, may I suggest looking at [Lua records] in PowerDNS Authoritative instead?

[Lua records]: https://doc.powerdns.com/authoritative/lua-records/
